src/postings/recorder.rs

```diff
@@ -171,13 +187,27 @@ impl Recorder for TermFrequencyRecorder {
         buffer_lender: &mut BufferLender,
         serializer: &mut FieldSerializer<'_>,
         heap: &MemoryArena,
+        docid_map: Option<&DocidMapping>,
     ) {
         let buffer = buffer_lender.lend_u8();
         self.stack.read_to_end(heap, buffer);
         let mut u32_it = VInt32Reader::new(&buffer[..]);
-        while let Some(doc) = u32_it.next() {
-            let term_freq = u32_it.next().unwrap_or(self.current_tf);
-            serializer.write_doc(doc as u32, term_freq, &[][..]);
+        if let Some(docid_map) = docid_map {
+            let mut docid_and_tf = vec![];
+            while let Some(old_docid) = u32_it.next() {
+                let term_freq = u32_it.next().unwrap_or(self.current_tf);
+                docid_and_tf.push((docid_map.get_new_docid(old_docid), term_freq));
+            }
+            docid_and_tf.sort_unstable_by_key(|&(docid, _)| docid);
```

Comment:
some comment as a above. I suspect the performance impact is high. especially considering all the resize.